src/System.Private.ServiceModel/tests/Common/Infrastructure/src/BridgeClientAuthenticationManager.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+// 
+
+using System.Net;
+
+namespace Infrastructure.Common
+{
+    public static class BridgeClientAuthenticationManager
+    {
+        private const string AuthenticationResourceName = "WcfService.TestResources.AuthenticationResource";
+        private const string UsernameKeyName = "authUsername";
+        private const string PasswordKeyName = "authPassword";
+        private static readonly object s_authLock = new object();
+        private static NetworkCredential s_networkCredential;
+
+        public static NetworkCredential NetworkCredential
+        {
+            get
+            {
+                if (s_networkCredential == null)
+                {
+                    lock (s_authLock)
+                    {
+                        if (s_networkCredential == null)
+                        {
+                            var response = BridgeClient.MakeResourceGetRequest(AuthenticationResourceName, null);
+                            var credential = new NetworkCredential();
+                            credential.UserName = response[UsernameKeyName];
+                            credential.Password = response[PasswordKeyName];
```

Comment:
Nit: possibly worth an exception if the username and password are missing.  Your call.  I'm trying to think how best to diagnose auth failures that show up in the labs and looking for good ideas.